BrowserKit/Sources/ToolbarKit/NavigationToolbar/BrowserNavigationToolbar.swift

```diff
@@ -5,6 +5,10 @@
 import UIKit
 import Common
 
+public protocol BrowserNavigationToolbarDelegate: AnyObject {
+    func configureCFR(for button: ToolbarButton)
```

Comment:
curious - I notice that with the redux architecture, we replace the use of delegates since it provides a single direction flow that we can dispatch the action and then receive the updated state in `newState`. Curious to understand why configure this delegate as opposed to dispatching an action of `showCFR` and the handling the new state in the BrowserViewController. 

In BVC newState method, we are calling `setupMiddleButtonStatus` which called the configuration for the CFR in the old code. Is this something that was difficult to use for the new toolbar code?